src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,442 @@
+ * within the packages defined by the method getPackageNames() (default {@link org.broadinstitute.hellbender.tools.walkers.annotator}).
+ * Also handles integrating annotation specific arguments from the command line with tool specified defaults.
+ *
+ * Unlike {@link GATKReadFilterPluginDescriptor} annotation order is not important and thus argument order is not guaranteed to be
+ * preserved in all cases, especially when group annotations are involved.
+ *
+ * An alternative method for discovering annotations is ClassUtils.knownSubInterfaceSimpleNames(Annotation.class), which can
+ * be invoked in absence of command line inputs.
+ *
+ * NOTE: this class enforces that annotations with required arguments must see their arguments, yet this is not currently tested
+ *       as no such annotations exist in the GATK.
+ */
+public class GATKAnnotationPluginDescriptor  extends CommandLinePluginDescriptor<Annotation> {
+    //TODO this should be a configurable option or otherwise exposed to the user when configurations are fully supported.
+    private static final String pluginPackageName = "org.broadinstitute.hellbender.tools.walkers.annotator";
+    private static final Class<?> pluginBaseClass = org.broadinstitute.hellbender.tools.walkers.annotator.Annotation.class;
+
+    protected transient Logger logger = LogManager.getLogger(this.getClass());
+
+    @ArgumentCollection
+    private final GATKAnnotationArgumentCollection userArgs;
+
+    // Map of Annotation (simple) class names to the corresponding discovered plugin instance
+    private final Map<String, Annotation> allDiscoveredAnnotations = new HashMap<>();
+
+    // Map of Annotation (simple) class names to the corresponding default plugin instance.
+    // We keep the actual instances in case they have any additional state provided by the tool
+    // when they were created
+    private final Map<String, Annotation> toolDefaultAnnotations = new HashMap<>();
+    private final Set<String> toolDefaultGroups = new HashSet<>();
+
+    // Set of predecessor annotations for which we've seen arguments exist either as a tool default or be supplied by the user
+    // (eg. InbreedingCoeff if we see "--founderID" on the command line)
+    private final Set<String> requiredPredecessors = new HashSet<>();
+
+    // Map of annotation group name to list of annotations flagged with that group. The key here the simple name of the interface
+    // that describes the annotation group.
+    private final Map<String, Map<String, Annotation>> discoveredGroups = new HashMap<>();
+
+    // Annotation arguments that are shared and thus must be part of the plugin descriptor
+    @Argument(fullName = "founderID", shortName = "founderID", doc="Samples representing the population \"founders\"", optional=true)
+    private List<String> founderIds;
+
+    @Argument(fullName = "pedigree", shortName = "ped", doc="Pedigree file for determining the population \"founders\"", optional=true)
```

Comment:
I think we're not using short names like "ped", just reusing the long name ? Also add a constant for this and use in the tests.

Reply:
Actually, I notice there is already a constant for this in std args ("ped") PEDIGREE_FILE_SHORT_NAME that is used by SelectVariants and CalculateGenotypePosteriors. Should we re-use that ? Will that cause any downstream clashes ?

Reply:
bumped to the common argument. And I have checked in the next branch for places where there is a conflict and it's currently more or less separable.